### PR TITLE
Migrate model plugin API and compiler to onos-config

### PR DESCRIPTION
**Pull request by kuujo**

### CODE REVIEW

**Review by tomikazi — APPROVED** [review]

File: model/registry/registry.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		return model, errors.NewInvalid(err.Error())
+	}
+	if model.Name == "" || model.Version == "" {
+		return model, errors.NewInvalid("'%s' is not a valid model descriptor", path)
+	}
+	return model, nil
+}
+
+// GetPath :
+func GetPath(dir, target, replace string) (string, error) {
```
Comment (SeanCondon) [inline model/registry/registry.go:171]: this was unused in the old version

File: test/model/test@2020-11-18.yang [hunk 2]
```diff
@@ -0,0 +1,39 @@
+module test {
+  namespace "http://opennetworking.org/oran/test";
+  prefix t1;
+
+  organization
+    "Open Networking Foundation.";
+  contact
+    "Adib Rastegarnia";
```
Comment (SeanCondon) [inline test/model/test@2020-11-18.yang:8]: I wrote this !

**Review by SeanCondon — APPROVED** [review]

**Review by tomikazi — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -66,6 +76,7 @@ kind-only: # @HELP deploy the image without rebuilding first
 kind-only:
 	@if [ "`kind get clusters`" = '' ]; then echo "no kind cluster found" && exit 1; fi
 	kind load docker-image --name ${KIND_CLUSTER_NAME} ${DOCKER_REPOSITORY}onos-config:${ONOS_CONFIG_VERSION}
+	kind load docker-image --name ${KIND_CLUSTER_NAME} ${DOCKER_REPOSITORY}config-model-registry:${ONOS_CONFIG_VERSION}
```
Comment (ray-milkey) [inline Makefile:79]: You'll need to make a corresponding change to build/bin/push-images to publish the new image

File: Makefile [hunk 4]
```diff
@@ -66,6 +76,7 @@ kind-only: # @HELP deploy the image without rebuilding first
...
 
 all: build images
```
Comment (ray-milkey) [inline Makefile:82]: You'll have to add the new image to the publish target:

./../build-tools/publish-version ${VERSION} onosproject/config-model-registry

**Review by ray-milkey — CHANGES_REQUESTED** [review]

**Comment by kuujo** [comment]
Superseded by sidecar based plugins

**Pull request closed without merging**